src/main/java/seedu/address/logic/parser/AddStudentCommandParser.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PARENT_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PARENT_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENT_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENT_PHONE;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddStudentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.ParentName;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.StudentName;
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddStudentCommandParser implements Parser<AddStudentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddStudentCommand parse(String args) throws ParseException {
```

Comment:
I think there may be a need to handle the case where `add -s` is entered, either here or maybe in the `AddressBookParser::parseCommand`. Currently it throws an error due to array index out of range (I believe it's looking for an input flag).